src/life.rs

```diff
@@ -1,702 +1,20 @@
-use crate::cell::{Cell, CellState};
+use crate::grid::Grid;
 
 #[derive(Clone, Debug)]
 pub struct LifeState {
-    pub grid: Vec<Vec<Cell>>,
+    pub grid: Grid,
 }
 
 impl LifeState {
-    pub fn new(seed: Vec<Vec<Cell>>) -> Self {
-        Self { grid: seed }
+    pub fn new(grid: &Grid) -> Self {
+        Self { grid: grid.clone() }
     }
 
-    fn count_neighbors(&self, row_index: u32, col_index: u32) -> u32 {
-        let mut neighbors_count = 0;
+    pub fn step(&mut self) -> LifeState {
```

Comment:
If the only function of the `LifeState` struct is to have the `Grid` struct and calls `update()`, we could delete this struct and directly use `Grid`

Reply:
Before there was only `LifeState`, the grid was part of it. I modularized the grid to unclutter `LifeState` because I was planning on adding a few things to it (a generation counter, firstly). Haven't had the time yet. I'll keep `LifeState`. Makes sense?

Reply:
Ah in that case is all fine, just keep it

Reply:
thx :)